scaleway/data_source_image.go

```diff
@@ -21,17 +24,26 @@ func dataSourceScalewayImage() *schema.Resource {
 				Description: "exact name of the desired image",
 			},
 			"name_filter": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				ForceNew:    true,
-				Description: "partial name of the desired image to filter with",
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Description:   "partial name of the desired image to filter with",
+				ConflictsWith: []string{"most_recent"},
 			},
 			"architecture": {
 				Type:        schema.TypeString,
 				Required:    true,
 				ForceNew:    true,
 				Description: "architecture of the desired image",
 			},
+			"most_recent": {
+				Type:          schema.TypeBool,
+				Required:      false,
+				Default:       false,
```

Comment:
_minor_ you don't need to specify either of these, since they should be implied (false is the default value for a bool / required will be false since it's optional)

Reply:
gotcha.